pkg/tags/tags.go

```diff
@@ -103,32 +103,6 @@ func (tag Tag) Validate() error {
 	return checkValue(tag.Value)
 }
 
-// MarshalXML encodes to XML data.
-func (tag Tag) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
```

Comment:
`Tag` type was added primarily for `minio/pkg/bucket/lifecycle`. I would recommend to move `Tag` type to `minio/pkg/bucket/lifecycle` and `tags` to deal with `TagSet`